components/tools/OmeroWeb/omeroweb/webclient/webclient_gateway.py

```diff
@@ -2397,16 +2397,36 @@ def __prepare__(self, **kwargs):
         if 'link' in kwargs:
             self.link = 'link' in kwargs and kwargs['link'] or None
 
-    """
-    This override standard omero.gateway.ImageWrapper.getChannels
-    and catch exceptions.
-    """
+    def getPixelSizeUnits(self):
+        """
+        We use the pixelSizeX to find the appropriate units.
+        By default, we get the actual size in Microns, then use the
+        lengthunit filter to pick the most suitable size (same logic
+        is used for doing the length conversions).
+        However, if unit can't be converted, then we just return the
+        current unit's symbol.
+        """
+        from webgateway.templatetags.common_filters import lengthunit
+        try:
+            size = self.getPixelSizeX(units="MICROMETER")
```

Comment:
Why are we defaulting to micrometres?  If the data is using another unit, why are we not defaulting to that and then finding a suitable size only if it's too far our for the value being used.  What happens for non-metric units e.g. inch?

Basically, I'm asking why we are imposing this policy in the client?  If the image was using microns, or was converted to microns for historical reasons, that will already be microns in the database, making the default here moot.  But it does mean for new files using non-micron units, we are overriding what should be the default display unit.